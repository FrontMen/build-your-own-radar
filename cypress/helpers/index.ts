export const dataCy = (value: string) => cy.get(`[data-cy=${value}]`);
export const dataTestId = (value: string) => cy.get(`[data-testid=${value}]`);

export const hexToRgb = (hex: string) => {
  const bigint = parseInt(hex.replace('#', ''), 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
}
