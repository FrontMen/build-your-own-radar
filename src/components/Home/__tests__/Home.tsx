import React from 'react';
import { Home } from '../index';
import { withAllProviders } from '../../../../test/helpers';

describe('Home', () => {
  const { getByTestId } = withAllProviders(<Home />);

  it('should render correctly', () => {
    const container = getByTestId('home-title');
    expect(container).toHaveTextContent('FM Tech Radar');
  });
});
