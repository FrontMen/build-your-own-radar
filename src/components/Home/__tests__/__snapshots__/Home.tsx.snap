// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly when content is there 1`] = `
.c0 {
  grid-area: main;
  background-color: #fff;
  padding: 16px;
}

.c17 {
  font-family: Montserrat,san-serif;
  font-weight: 700;
  line-height: 1.15;
  font-size: 1.5em;
  text-transform: capitalize;
}

.c9 {
  width: 100%;
  max-width: 800px;
  min-width: 280px;
  height: auto;
  margin: 16px auto 32px;
  position: relative;
}

.c10 {
  text-align: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  right: -10%;
  bottom: -3%;
  width: 220px;
  padding: 0 20px;
  text-align: center;
  font-weight: bold;
  font-family: Montserrat,san-serif;
  pointer-events: none;
  -webkit-transform: rotateZ( -35deg );
  -ms-transform: rotateZ( -35deg );
  transform: rotateZ( -35deg );
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  left: -15%;
  bottom: 5%;
  width: 220px;
  padding: 0 20px;
  text-align: center;
  font-weight: bold;
  font-family: Montserrat,san-serif;
  pointer-events: none;
  -webkit-transform: rotateZ( 35deg );
  -ms-transform: rotateZ( 35deg );
  transform: rotateZ( 35deg );
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  left: -15%;
  width: 220px;
  padding: 0 20px;
  text-align: center;
  font-weight: bold;
  font-family: Montserrat,san-serif;
  pointer-events: none;
  -webkit-transform: rotateZ( -34deg );
  -ms-transform: rotateZ( -34deg );
  transform: rotateZ( -34deg );
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 5%;
  right: -15%;
  width: 220px;
  padding: 0 20px;
  text-align: center;
  font-weight: bold;
  font-family: Montserrat,san-serif;
  pointer-events: none;
  -webkit-transform: rotateZ( 36deg );
  -ms-transform: rotateZ( 36deg );
  transform: rotateZ( 36deg );
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  padding: 8px 0;
  margin-bottom: 20px;
}

.c6 {
  -webkit-flex: 0 0 1.5em;
  -ms-flex: 0 0 1.5em;
  flex: 0 0 1.5em;
  height: 2em;
  width: 2em;
  margin: 5px;
}

.c8 {
  border: none;
  border-bottom: 2px solid;
  border-color: #767676;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  outline: none;
  -webkit-appearance: none;
}

.c8:focus {
  border-color: #526de4;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.c1 {
  margin: auto;
  margin-bottom: 8px;
  max-width: 60em;
}

.c15 {
  display: grid;
  grid-template-columns: repeat(1fr);
  grid-gap: 32px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 8px;
}

.c3 {
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  margin: 0;
  margin-bottom: 16px;
}

.c18 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 0.9em 1.25em;
  padding-right: 3.5em;
  color: #fff;
  background-color: #526de4;
  border-color: #526de4;
}

.c18 img {
  position: absolute;
  top: 15px;
  right: 15px;
  width: 1.4em;
  height: 1.1em;
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.23,1,0.32,1);
  -webkit-transition: transform 0.3s cubic-bezier(0.23,1,0.32,1);
  transition: transform 0.3s cubic-bezier(0.23,1,0.32,1);
}

.c18:hover img {
  -webkit-transform: translateX(5px);
  -ms-transform: translateX(5px);
  transform: translateX(5px);
}

.c2 {
  font-family: Montserrat,san-serif;
  font-weight: 700;
  line-height: 1.15;
  font-size: 1.5em;
  text-transform: capitalize;
  color: #526de4;
}

.c5 {
  max-width: 60em;
  margin: 0 auto;
  margin-bottom: 20px;
}

@media (min-width:481px) {
  .c0 {
    padding: 32px;
  }
}

@media (min-width:481px) {
  .c17 {
    font-size: 1.5em;
    margin-top: 0;
    margin-bottom: 16px;
  }
}

@media (min-width:481px) {
  .c9 {
    max-width: 80%;
    margin-bottom: 16px;
  }
}

@media (min-width:1024px) {
  .c9 {
    max-width: 50%;
    margin-bottom: 64px;
  }
}

@media (min-width:1024px) {
  .c10 {
    position: -webkit-sticky;
    position: sticky;
    top: 0;
  }
}

@media (min-width:768px) {
  .c15 {
    grid-template-columns: repeat(auto-fit,minmax(25em,1fr));
  }
}

@media (min-width:768px) {
  .c16 {
    width: 100%;
    margin-bottom: 32px;
  }
}

@media (min-width:768px) {
  .c3 {
    height: 110px;
  }
}

@media (min-width:481px) {
  .c2 {
    font-size: 1.5em;
    margin-top: 0;
    margin-bottom: 16px;
  }
}

@media (min-width:768px) {
  .c5 {
    margin-top: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="home-intro"
    >
      <h2
        class="c2"
        data-testid="home-intro-title"
      >
        About
      </h2>
      <p
        class="c3"
        data-testid="home-intro-content"
      >
        The Intracto Technology Radar displays our assessments about the risks and rewards of existing and upcoming technologies, tools and techniques we use to develop digital applications. It is a living document which is periodically updated by our experts and displays the evolution over time how we use these technologies and techniques.
      </p>
    </div>
    <div
      class="c4 c5 home-search"
      data-testid="subnav-search-container"
    >
      <svg
        class="c6"
        data-testid="search-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
        />
      </svg>
      <div
        class="c7"
      >
        <input
          class="c8"
          data-testid="search-input"
          placeholder="Search..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="c9"
    >
      <div
        class="c10"
      >
        <svg
          data-testid="graph"
        />
      </div>
      <div
        class="c11"
        data-testid="graph-tooltip-mobile"
      >
        Frameworks, CMS & Programmeertalen
      </div>
      <div
        class="c12"
        data-testid="graph-tooltip-mobile"
      >
        Tooling en testing
      </div>
      <div
        class="c13"
        data-testid="graph-tooltip-mobile"
      >
        Platforms, infrastructure & Data
      </div>
      <div
        class="c14"
        data-testid="graph-tooltip-mobile"
      >
        Technieken
      </div>
    </div>
    <div
      class="c15"
      data-testid="home-quadrants-wrapper"
    >
      <div
        class="c16"
        data-testid="home-quadrant-0-container"
      >
        <h2
          class="c17"
          data-testid="home-quadrant-0-title"
        >
          Tooling & Testing
        </h2>
        <p
          class="c3"
          data-testid="home-quadrant-0-content"
        >
          Successful application development relies on the use of the best suited frameworks, cms and programming languages. This quadrant displays the technologies we use for application development.
        </p>
        <a
          class="c18"
          data-testid="home-quadrant-0-link"
          href="/frameworks-and-lang"
        >
          Overview of 
          Frameworks, CMS & programming languages
          <img
            alt="right-arrow"
            src="arrow-right.svg"
          />
        </a>
      </div>
      <div
        class="c16"
        data-testid="home-quadrant-1-container"
      >
        <h2
          class="c17"
          data-testid="home-quadrant-1-title"
        >
          Tooling & Testing
        </h2>
        <p
          class="c3"
          data-testid="home-quadrant-1-content"
        >
          To ensure effective application development and the delivery of applications with optimal UX, performance, maintainability and security, we use a range of tooling and testing frameworks.
        </p>
        <a
          class="c18"
          data-testid="home-quadrant-1-link"
          href="/tooling-and-testing"
        >
          Overview of 
          Tooling & Testing
          <img
            alt="right-arrow"
            src="arrow-right.svg"
          />
        </a>
      </div>
      <div
        class="c16"
        data-testid="home-quadrant-2-container"
      >
        <h2
          class="c17"
          data-testid="home-quadrant-2-title"
        >
          Tooling & Testing
        </h2>
        <p
          class="c3"
          data-testid="home-quadrant-2-content"
        >
          Applications can be build on or served from platforms (e.g. SaaS) and/ or need platforms and infrastructure for hosting (e.g. cloud). Applications often rely on other applications to manage their data and message flows such as databases, caching applications or service/ message busses.
        </p>
        <a
          class="c18"
          data-testid="home-quadrant-2-link"
          href="/platforms-infra-and-data"
        >
          Overview of 
          Platforms, infrastructure & data
          <img
            alt="right-arrow"
            src="arrow-right.svg"
          />
        </a>
      </div>
      <div
        class="c16"
        data-testid="home-quadrant-3-container"
      >
        <h2
          class="c17"
          data-testid="home-quadrant-3-title"
        >
          Tooling & Testing
        </h2>
        <p
          class="c3"
          data-testid="home-quadrant-3-content"
        >
          The design, development, hosting, maintenance and monitoring of applications and infrastructure can be complex. To ensure optimal efficiency and quality for all of these topics, we use a variety of proven techniques and market standards.
        </p>
        <a
          class="c18"
          data-testid="home-quadrant-3-link"
          href="/technique"
        >
          Overview of 
          Techniques
          <img
            alt="right-arrow"
            src="arrow-right.svg"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
